test/subdomainregistrar/TestRentalSubdomainRegistrar.js

```diff
@@ -0,0 +1,622 @@
+const { ethers } = require('hardhat')
+const { utils, BigNumber: BN } = ethers
+const { use, expect } = require('chai')
+const { solidity } = require('ethereum-waffle')
+const n = require('eth-ens-namehash')
+const namehash = n.hash
+const { FUSES } = require('../test-utils/ens')
+const { deploy } = require('../test-utils/contracts')
+
+const { CANNOT_UNWRAP, PARENT_CANNOT_CONTROL } = FUSES
+
+use(solidity)
+
+const labelhash = (label) => utils.keccak256(utils.toUtf8Bytes(label))
+const ROOT_NODE =
+  '0x0000000000000000000000000000000000000000000000000000000000000000'
+
+const EMPTY_BYTES32 =
+  '0x0000000000000000000000000000000000000000000000000000000000000000'
+const EMPTY_ADDRESS = '0x0000000000000000000000000000000000000000'
+const MAX_EXPIRY = 2n ** 64n - 1n
```

Comment:
What's the n here represent? Referring to line 5 it does not a seem to be a number

Reply:
2*n ** 64 * n - 1 * n work from my end

Reply:
Hey @yayashuxue , 

Will you be pushin a production ready version of the registrar soon?

Reply:
@yayashuxue will you be pushing a production ready version soon?

Reply:
n represents big numbers in Javascript, you can read it without the n